### PR TITLE
Add UT for cl_khr_external_memory related functions

### PR DESCRIPTION
**Pull request by kamilwiatrowskimobica**

### CODE REVIEW

**Review by bashbaug — APPROVED** [review]
LGTM, needs the merge conflict resolved.

File: tests/test_openclhpp.cpp [hunk 1]
```diff
@@ -3833,4 +3880,103 @@ void testSemaphoreRetain(void) {}
 void testSemaphoreRelease(void) {}
 #endif // cl_khr_semaphore
 
+/****************************************************************************
+ * Tests for cl_khr_external_memory
+ ****************************************************************************/
+static cl_int clEnqueueAcquireExternalMemObjectsKHR_testEnqueueAcquireExternalMemObjects(
```
Comment (bashbaug) [inline tests/test_openclhpp.cpp:3886]: I think these tests are all OK as-is, but if you want to be on the safe side you could consider wrapping them all inside of one big ifdef for `cl_khr_external_memory` similar to what was done for the `cl_khr_semaphore` tests above.

File: tests/test_openclhpp.cpp [hunk 2]
```diff
@@ -1294,6 +1302,45 @@ void testBufferConstructorQueueIterator(void)
...
+
+    return make_mem(0);
+}
+
+void testBufferWithProperties(void)
+{
+#if CL_HPP_TARGET_OPENCL_VERSION >= 300
+    clCreateBufferWithProperties_StubWithCallback(clCreateBufferWithProperties_testBufferWithProperties);
+
+    VECTOR_CLASS<cl_mem_properties> props{CL_DEVICE_HANDLE_LIST_KHR};
```
[the remark this answers is not in the document]
Comment (kamilwiatrowskimobica) [reply tests/test_openclhpp.cpp]: Done using option 2.

File: tests/test_openclhpp.cpp (hunk 1 above)
Comment (kamilwiatrowskimobica) [reply tests/test_openclhpp.cpp:3887]: Done.

**Comment by bashbaug** [comment]
Thanks!  I'm going to go ahead and merge this one since it's been open a while and it seems relatively non-controversial.

**Pull request merged**